### DISCUSSION TITLE
Zod preprocessor ignores default value

### DISCUSSION THREAD

**Discussion by lvndry** [colinhacks/zod#3017] [Q&A] [answered]
* Use case
I'm trying to get the default value of a preprocessed input.

I tried

```typescript
const test_value = z.preprocess(stringParser, z.string().default("test_key"));
const parsed_value = test_value.parse(undefined); // => void
```

and 

```typescript
const test_value = z.preprocess(stringParser, z.string().default("test_key")).default("test_key");
const parsed_value = test_value.parse(undefined); // => void
```

* Current behavior

The default value is not computed and `z.parse` returns nothing

* Expected behavior

Default value should be returned if input couldn't be preprocessed

**Comment by JacobWeisenburger** [comment, marked as answer]
Please send a full reproducible code example.

![image](https://github.com/colinhacks/zod/assets/31667350/cb015eb7-d0af-48be-afdd-cb763f04ffef)

**Comment by lvndry** [comment]
Actually I found that out that it's because I had to make sure that value returned by the `stringParser` is `undefined` and not `""`.
Closing the issue
